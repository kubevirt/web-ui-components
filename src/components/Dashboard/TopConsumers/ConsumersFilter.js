import React from 'react';
import PropTypes from 'prop-types';

export const ConsumersFilter = ({ children }) => <div className="kubevirt-consumers__filters">{children}</div>;

ConsumersFilter.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
