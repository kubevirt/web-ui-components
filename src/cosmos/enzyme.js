import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
