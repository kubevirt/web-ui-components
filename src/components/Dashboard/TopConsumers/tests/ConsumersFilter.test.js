import React from 'react';
import { shallow } from 'enzyme';

import { ConsumersFilter } from '../ConsumersFilter';
import { default as ConsumersFilterFixtures } from '../fixtures/ConsumersFilter.fixture';

const testConsumersFilter = () => <ConsumersFilter {...ConsumersFilterFixtures.props} />;

describe('<ConsumersFilter />', () => {
  it('renders correctly', () => {
    const component = shallow(testConsumersFilter());
    expect(component).toMatchSnapshot();
  });
});
