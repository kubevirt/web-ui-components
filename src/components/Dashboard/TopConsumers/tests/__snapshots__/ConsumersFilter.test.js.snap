// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsumersFilter /> renders correctly 1`] = `
<div
  className="kubevirt-consumers__filters"
>
  <Dropdown
    choices={
      Array [
        "choice1",
        "choice2",
      ]
    }
    disabled={false}
    id="id"
    onChange={[Function]}
    value="fooValue"
    withTooltips={false}
  />
</div>
`;
